lib/use-cases/GetEvidenceStoreUrl.js

```diff
@@ -1,10 +1,18 @@
 module.exports = ({ gateways: { evidenceStore } }) => {
   return async dropboxId => {
-    const uploadOptions = await evidenceStore.createUploadUrl(dropboxId);
-    return {
-      documentId: uploadOptions.documentId,
-      url: uploadOptions.url,
-      fields: uploadOptions.fields
-    };
+    const documents = await evidenceStore.getByDropboxId(dropboxId);
+    const unusedDocuments = documents.filter(d => !d.filename);
+
+    if (unusedDocuments.length > 0) {
+      const { documentId, url, fields } = await evidenceStore.getUploadUrl(
+        unusedDocuments[0].id
+      );
+      return { documentId, url, fields };
+    } else {
+      const { documentId, url, fields } = await evidenceStore.createUploadUrl(
+        dropboxId
+      );
+      return { documentId, url, fields };
+    }
```

Comment:
Could write this bit without the `else` since we already returned from inside of the `if` block, might be a bit cleaner to read?

Reply:
I think I'd have to give the vars different names?

Reply:
Have updated it upstream

`module.exports = ({ gateways: { evidenceStore } }) => {
  return async dropboxId => {
    const documents = await evidenceStore.getByDropboxId(dropboxId);
    const unusedDocuments = documents.filter(d => !d.filename);

    const { documentId, url, fields } =
      unusedDocuments.length > 0
        ? await evidenceStore.getUploadUrl(unusedDocuments[0].id)
        : await evidenceStore.createUploadUrl(dropboxId);

    return { documentId, url, fields };
  };
};`